src/main/kotlin/app/revanced/patches/shared/settings/preference/impl/SwitchPreference.kt

```diff
@@ -10,17 +10,17 @@ import org.w3c.dom.Element
  *
  * @param key The key of the switch.
  * @param title The title of the switch.
- * @param default The default value of the switch.
  * @param summaryOn The summary to show when the preference is enabled.
  * @param summaryOff The summary to show when the preference is disabled.
  * @param userDialogMessage The message to show in a dialog when the user toggles the preference.
+ * @param default The default value of the switch.
  */
 internal class SwitchPreference(
     key: String, title: StringResource,
-    val default: Boolean = false,
     val summaryOn: StringResource? = null,
     val summaryOff: StringResource? = null,
-    val userDialogMessage: StringResource? = null
+    val userDialogMessage: StringResource? = null,
+    val default: Boolean? = null,
```

Comment:
Why not:

```kotlin
val default: Boolean = false,
```

Reply:
I may not understand this correctly. I believe, Android treats preferences that do not specify the `default` attribute as false by default. This would mean, an if check on this parameter would suffice, to decide weather to write the attribute to the XML document, or not.